src/Main.ps1

```diff
@@ -1214,6 +1217,9 @@ function Invoke-Pester {
             }
         }
 
+        # go back to original CWD
+        & $SafeCommands['Pop-Location'] -StackName 'Pester'
```

Comment:
I think doing this in non-stateful way with a variable would be better. This is expensive, Pester-in-Pester can possibly push on that stack, and simply doing` if ($null -ne $script:currentPwd) { cd... }` seems like less problematic code.

Reply:
So I guess we got a choice between:
- Push/Pop, but change StackName to `Pester_<$state.Id>` and add a guid to `New-PesterState`. RunId might be useful in the future.
- Add `$state.InitialDirectory = $pwd.Path` and only `Set-Location` at the end

Reply:
The last option is used in other places as well, where we read variable that was defined in parent scope, to e.g. figure out if we are running Pester in Pester. As long as the scope that returns to the final folder is at the same scope level or below, this is probably the best choice.